homeassistant/components/statistics/sensor.py

```diff
@@ -141,32 +165,46 @@ def __init__(
         self._source_entity_id = source_entity_id
         self.is_binary = self._source_entity_id.split(".")[0] == "binary_sensor"
         self._name = name
-        self._available = False
         self._state_characteristic = state_characteristic
         self._samples_max_buffer_size = samples_max_buffer_size
         self._samples_max_age = samples_max_age
         self._precision = precision
         self._quantile_intervals = quantile_intervals
         self._quantile_method = quantile_method
+        self._value = None
         self._unit_of_measurement = None
+        self._available = False
         self.states = deque(maxlen=self._samples_max_buffer_size)
         self.ages = deque(maxlen=self._samples_max_buffer_size)
-        self.attr = {
-            STAT_COUNT: 0,
-            STAT_TOTAL: None,
-            STAT_MEAN: None,
-            STAT_MEDIAN: None,
-            STAT_STANDARD_DEVIATION: None,
-            STAT_VARIANCE: None,
-            STAT_MIN_VALUE: None,
-            STAT_MAX_VALUE: None,
-            STAT_MIN_AGE: None,
-            STAT_MAX_AGE: None,
-            STAT_CHANGE: None,
-            STAT_AVERAGE_CHANGE: None,
-            STAT_CHANGE_RATE: None,
-            STAT_QUANTILES: None,
+        self.attributes = {
+            STAT_AGE_COVERAGE_RATIO: STATE_UNKNOWN,
+            STAT_BUFFER_USAGE_RATIO: STATE_UNKNOWN,
+            STAT_SOURCE_VALUE_VALID: STATE_UNKNOWN,
         }
+        self.statistics = {
```

Comment:
We don't need to make a copy of this map in each instance.

Maybe just do something like:
```py
self._state_characteristic_fn = getattr(self, f"_stat_{self._state_characteristic})
```

Reply:
I went down that route before but was greeted by some linter warnings, can't remember which.
Happy to try again.

Reply:
It's fine to silence the linter in that case.